core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala

```diff
@@ -3487,8 +3487,12 @@ class ReplicaManagerTest {
         assertTrue(replicaManager.logManager.getLog(tp0).isDefined)
       }
       if (enableRemoteStorage) {
-        verify(mockRemoteLogManager, times(1))
-          .stopPartitions(ArgumentMatchers.eq(Collections.singleton(tp0)), ArgumentMatchers.eq(false), any())
+        if (throwIOException) {
+          verify(mockRemoteLogManager, times(0)).stopPartitions(any(), any())
```

Comment:
Could you explain more about why we don't call `stopPartitions` when `throwIOException` now? My understanding is this should be a bug previously that we didn't filter out the error partition. Is that right?

Reply:
yes, there was a bug. Previously, we were calling the RLM#stopPartitions twice one with `delete` set as true and false. The error partition was by-passing through the second call.

Reply:
Good catch. Thanks.